lib/Client.js

```diff
@@ -3487,22 +3487,30 @@ class Client extends EventEmitter {
      * Get a member that is part of a thread channel
      * @arg {String} channelID The ID of the thread channel
      * @arg {String} userID The ID of the user
-     * @arg {Boolean} withMember Whether to include the guild member object within the response
+     * @arg {Boolean} [withMember] Whether to include the guild member object within the response
      * @returns {Promise<ThreadMember>}
      */
     getThreadMember(channelID, userID, withMember) {
-        return this.requestHandler.request("GET", Endpoints.THREAD_MEMBER(channelID, userID), {
+        return this.requestHandler.request("GET", Endpoints.THREAD_MEMBER(channelID, userID), true, {
             with_member: withMember
         }).then((threadMember) => new ThreadMember(threadMember, this));
     }
 
     /**
      * Get a list of members that are part of a thread channel
      * @arg {String} channelID The ID of the thread channel
+     * @arg {Object} [options] Options for the request
+     * @arg {String} [options.after] Get members after this user ID
+     * @arg {Number} [options.limit=100] The maximum number of thread members to fetch (1-100)
+     * @arg {Boolean} [options.withMember] Whether to include the guild member object for each member of the thread
      * @returns {Promise<Array<ThreadMember>>}
      */
-    getThreadMembers(channelID) {
-        return this.requestHandler.request("GET", Endpoints.THREAD_MEMBERS(channelID), true).then((members) => members.map((member) => new ThreadMember(member, this)));
+    getThreadMembers(channelID, options) {
+        return this.requestHandler.request("GET", Endpoints.THREAD_MEMBERS(channelID), true, {
+            after: options.after,
+            limit: options.limit,
+            with_member: options.withMember
+        }).then((members) => members.map((member) => new ThreadMember(member, this)));
```

Comment:
Need to snipe `id` into the member object otherwise cache cannot be upserted

Reply:
ThreadMember structure already handles that?

Reply:
You're thinking about the thread channel ID, not the member

Reply:
it handles the member too

Reply:
There is no `id` in the member object from the api, the guild members collection requires it to update the guild member.

![image](https://github.com/abalabahaha/eris/assets/24302436/78daf65b-de93-478e-af40-ab3145c15f31)

Reply:
oh mb forgot discord was acoustic. fixed ??